### PR TITLE
Make function names globally unique

### PR DESCRIPTION
**Pull request by FabianSperrle**
I am not sure what the status of this project is or what it's intended to be used for -- I am (ab)using it to allow non-developers to create some configurations in yaml and parsing that to code. Many of the created files will have the same format, and, importantly, might define functions with the same name.

I just spent a few hours debugging why nothing was working until it finally clicked. I had defined functions with the same name (in the python code part) in multiple yaml files. 

This MR modifies all functions to include a random uuid suffix, making them only accessible through the lambda in the rickled object.

Maybe this is helpful for somebody else as well.

### CODE REVIEW

**Comment by cuniculus-zipf** [comment]
Hi. Thanks for the contribution. It seems like a good idea, with no breaking changes. 

Will merge with next release

File: rickled/__init__.py [hunk 1]
```diff
@@ -108,10 +111,17 @@ def from_rickle(self, rickle, cls):
...
+            obj = cls(**args)
+        else:
+            obj = cls()
+
+        # print('the rickle', rickle)
+        # print('the rickle dict', rickle.dict())
 
         for name, value in rickle.dict().items():
             if isinstance(value, dict) and 'type' in value.keys():
+                # print(name, value)
```
Comment (cuniculus-zipf) [inline rickled/__init__.py:124]: @FabianSperrle after testing, please remove the commented lines.

File: rickled/__init__.py [hunk 2]
```diff
@@ -233,6 +250,7 @@ class BaseRickle:
     """
     def _iternalize(self, dictionary : dict, deep : bool, **init_args):
         for k, v in dictionary.items():
+            print(k,v)
```
Comment (cuniculus-zipf) [inline rickled/__init__.py:253]: @FabianSperrle review changes and remove prints. :)

File: rickled/__init__.py [hunk 3]
```diff
@@ -262,7 +281,11 @@ def __init__(self, base : Union[dict,str,TextIOWrapper,list] = None, deep : bool
             for file in base:
                 with open(file, 'r') as f:
                     stringed = f'{stringed}\n{f.read()}'
+        elif 'path' in init_args and os.path.isfile(os.path.join(init_args['path'], base)):
```
Comment (cuniculus-zipf) [inline rickled/__init__.py:284]: @FabianSperrle I'm not sure what the use case is here? Could you explain why this is useful?

**Review by cuniculus-zipf — CHANGES_REQUESTED** [review]
@FabianSperrle All in all, these look good. I just have some comments for you to look at, and then please add some unit tests, testing the use cases you've catered for.

File: rickled/__init__.py (hunk 3 above)
Comment (FabianSperrle) [reply rickled/__init__.py:284]: I am using the library in a FastAPI application and had issues with file paths being relative to the wrong files. In any case, as you have spotted, I continued with some quick and dirty development in my fork. I will set up a cleaned branch to merge into `dev` here in the next days.

File: rickled/__init__.py [hunk 4]
```diff
@@ -122,6 +132,13 @@ def from_rickle(self, rickle, cls):
                                             return_function=True, is_method=True, includes_self_reference=_includes_self_reference)
 
                     obj.__dict__.update({_name: partial(f, obj)})
+                if value['type'] == 'action':
+                    # print('loading action', value)
+                    from guidance_strategies.action import ConditionalGuidanceAction
```
Comment (cuniculus-zipf) [inline rickled/__init__.py:137]: @FabianSperrle This looks like something specific from your use case, and your own project classes. If so, please remove.

**Review by cuniculus-zipf — CHANGES_REQUESTED** [review]

**Comment by cuniculus-zipf** [comment]
@FabianSperrle I merged into `dev` and will update the requests I made.

**Comment by cuniculus-zipf** [comment]
@FabianSperrle May I add you to a list of contributors?

**Pull request merged**